discourse/discourse.php

```diff
@@ -0,0 +1,354 @@
+<?php
+
+/**
+ * Name: Discourse Mail Connector
+ * Description: Improves mails from Discourse in mailing list mode
+ * Version: 0.1
+ * Author: Michael Vogel <http://pirati.ca/profile/heluecht>
+ *
+ */
+//use DOMDocument;
+//use DOMXPath;
+use Friendica\App;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Logger;
+use Friendica\Core\PConfig;
+use Friendica\Core\Renderer;
+use Friendica\Core\Protocol;
+use Friendica\Database\DBA;
+use Friendica\Model\Contact;
+use Friendica\Util\XML;
+use Friendica\Content\Text\Markdown;
+use Friendica\Util\Network;
+use Friendica\Util\Strings;
+Use Friendica\Util\DateTimeFormat;
+
+/* Todo:
+ * - Obtaining API tokens to be able to read non public posts as well
+ * - Handling duplicates (possibly using some non visible marker)
+ * - Fetching missing posts
+ * - Fetch topic information
+ * - Support mail free mode when write tokens are available
+ * - Fix incomplete (relative) links (hosts are missing)
+*/
+
+function discourse_install()
+{
+	Hook::register('email_getmessage',        __FILE__, 'discourse_email_getmessage');
+	Hook::register('connector_settings',      __FILE__, 'discourse_settings');
+	Hook::register('connector_settings_post', __FILE__, 'discourse_settings_post');
+}
+
+function discourse_uninstall()
+{
+	Hook::unregister('email_getmessage',        __FILE__, 'discourse_email_getmessage');
+	Hook::unregister('connector_settings',      __FILE__, 'discourse_settings');
+	Hook::unregister('connector_settings_post', __FILE__, 'discourse_settings_post');
+}
+
+function discourse_settings(App $a, &$s)
+{
+	if (!local_user()) {
+		return;
+	}
+
+	$enabled = intval(PConfig::get(local_user(), 'discourse', 'enabled'));
+	$css = ($enabled ? '' : '-disabled');
+
+	$s .= '<span id="settings_discourse_inflated" class="settings-block fakelink" style="display: block;" onclick="openClose(\'settings_discourse_expanded\'); openClose(\'settings_discourse_inflated\');">';
```

Comment:
Pease use an addon template instead. The goal is to have no HTML code in PHP files.

Reply:
I did it like Twitter does. Where can I have a look at?

Reply:
https://github.com/friendica/friendica-addons/blob/b308f3bc0528b49a8ce0ccc9be21e047e6dff87d/advancedcontentfilter/advancedcontentfilter.php#L207-L235

Reply:
Even better: https://github.com/friendica/friendica-addons/blob/b308f3bc0528b49a8ce0ccc9be21e047e6dff87d/langfilter/langfilter.php#L43-L66